samples/cellular/modem_shell/src/location/location_shell.c

```diff
@@ -325,11 +370,40 @@ void location_ctrl_event_handler(const struct location_event_data *event_data)
 #endif
 #endif /* defined(CONFIG_LOCATION_SERVICE_EXTERNAL) */
 	case LOCATION_EVT_RESULT_UNKNOWN:
-		mosh_print("Location request completed, but the result is not known");
+		mosh_print("Location request completed, but the result is not known:");
+		mosh_print(
+			"  used method: %s (%d)",
+			location_method_str(event_data->method),
+			event_data->method);
 		break;
+#if defined(CONFIG_LOCATION_DATA_DETAILS)
 	case LOCATION_EVT_STARTED:
-		mosh_print("Location request has been started");
+		mosh_print("Location request has been started:");
+		mosh_print(
+			"  used method: %s (%d)",
+			location_method_str(event_data->method),
+			event_data->method);
 		break;
+	case LOCATION_EVT_FALLBACK:
+		mosh_print("Location request fallback has occurred:");
+		mosh_print(
+			"  failed method: %s (%d)",
+			location_method_str(event_data->method),
+			event_data->method);
+		mosh_print(
+			"  new method: %s (%d)",
+			location_method_str(event_data->fallback.next_method),
+			event_data->fallback.next_method);
+		mosh_print(
+			"  cause: %s",
+			(event_data->fallback.cause == LOCATION_EVT_TIMEOUT) ?
+			 "timeout" :
+			 (event_data->fallback.cause == LOCATION_EVT_ERROR) ?
+			  "error" :
```

Comment:
`timeout` and `error` could be on the same line with the condition, but I leave this up to you.
```
			(event_data->fallback.cause == LOCATION_EVT_TIMEOUT) ? "timeout" :
			(event_data->fallback.cause == LOCATION_EVT_ERROR) ? "error" :
			"unknown");
```

Reply:
I don't know how these are supposed to be intended and in some sense your proposal is "a bit wrong" as there is if-else structure without indentation. But I agree that's more readable so I've changed into it.